fictadvisor-api/src/v2/api/dtos/QueryAllCommentsDTO.ts

```diff
@@ -0,0 +1,29 @@
+import { validationOptionsMsg } from '../../utils/GLOBALS';
+import { ApiPropertyOptional } from '@nestjs/swagger';
+import { IsEnum, IsOptional, ValidateNested } from 'class-validator';
+import { Type } from 'class-transformer';
+import { QueryAllDTO } from '../../utils/QueryAllDTO';
+import { QuerySemesterDTO } from './QuerySemesterDTO';
+
+export enum SortCommentsEnum {
+  teacher = 'teacher',
+  subject = 'subject',
+  semester = 'semester',
+}
+
+export class QueryAllCommentsDTO extends QueryAllDTO {
+  @ApiPropertyOptional({
+    enum: SortCommentsEnum,
+    default: SortCommentsEnum.subject,
+  })
+  @IsEnum(SortCommentsEnum, validationOptionsMsg('Sort should be an enum'))
```

Comment:
we use pattern to name exceptions in such way: "Some value" must be a "type". So rename this exception according to the pattern